### PR TITLE
NUP-2397: rename TP* to TM*

### PR DESCRIPTION
**Pull request by marionleborgne**
Fixes https://github.com/numenta/nupic/issues/3556

Name changes:
* Rename `TPRegion` to `TMRegion`
* Rename `TP` to `BacktrackingTM`
* Rename `TP10X*` to `BacktrackingTMCPP`
* Rename `TPShim*` to `TMShim*`
* Rename generic usage of `TP / temporal pooling` to `TM / temporal memory`.

### CODE REVIEW

**Comment by marionleborgne** [comment]
cc @rhyolight @scottpurdy

**Comment by rhyolight** [comment]
@marionleborgne Will review today

File: examples/NuPIC Walkthrough.ipynb [hunk 1]
```diff
@@ -674,7 +674,7 @@
    "outputs": [],
    "source": [
     "# Step 1: create Temporal Pooler instance with appropriate parameters\n",
-    "tp = TP(numberOfCols=50, cellsPerColumn=2,\n",
+    "tm = TP(numberOfCols=50, cellsPerColumn=2,\n",
```
Comment (rhyolight) [inline examples/NuPIC Walkthrough.ipynb]: Is this constructor call still valid?

File: examples/tm/tm_constant_test.py [hunk 2]
```diff
@@ -68,36 +68,44 @@ def _createTps(numCols):
...
-  cppTp = TP10X2(numberOfCols=numCols, cellsPerColumn=cellsPerColumn,
-                  initialPerm=initialPerm, connectedPerm=connectedPerm,
-                  minThreshold=minThreshold, newSynapseCount=newSynapseCount,
-                  permanenceInc=permanenceInc, permanenceDec=permanenceDec,
-                  activationThreshold=activationThreshold,
-                  globalDecay=globalDecay, burnIn=1,
-                  seed=SEED, verbosity=VERBOSITY,
-                  checkSynapseConsistency=True,
-                  pamLength=1000)
+  cppTp = BacktrackingTMCPP(numberOfCols=numCols,
```
Comment (rhyolight) [inline examples/tm/tm_constant_test.py]: Change `cppTp` ==> `cppTm`

File: examples/tm/tm_constant_test.py [hunk 3]
```diff
@@ -68,36 +68,44 @@ def _createTps(numCols):
...
 
-  pyTp = TP(numberOfCols=numCols, cellsPerColumn=cellsPerColumn,
-             initialPerm=initialPerm, connectedPerm=connectedPerm,
-             minThreshold=minThreshold, newSynapseCount=newSynapseCount,
-             permanenceInc=permanenceInc, permanenceDec=permanenceDec,
-             activationThreshold=activationThreshold,
-             globalDecay=globalDecay, burnIn=1,
-             seed=SEED, verbosity=VERBOSITY,
-             pamLength=1000)
+  pyTp = BacktrackingTM(numberOfCols=numCols,
```
Comment (rhyolight) [inline examples/tm/tm_constant_test.py]: Change `pyTp` ==> `pyTm`

File: src/nupic/math/roc_utils.py [hunk 4]
```diff
@@ -229,7 +229,7 @@ def _test():
   1          0.3      0        0      0        1      1
   1          0.45     0        1      1        1      1
 
-  TruePos(TP)         1        2      2        3      3
+  TruePos(TM)         1        2      2        3      3
```
Comment (rhyolight) [inline src/nupic/math/roc_utils.py]: This `TM` should not be replaced because it means `TruePos`.

File: src/nupic/math/roc_utils.py [hunk 5]
```diff
@@ -238,7 +238,7 @@ def _test():
   The ROC curve is a plot of FPR on the x-axis and TPR on the y-axis. Basically,
   one can pick any operating point along this curve to run, the operating point
   determined by which threshold you want to use. By changing the threshold, you
-  tradeoff TP's for FPs.
+  tradeoff TM's for FPs.
```
Comment (rhyolight) [inline src/nupic/math/roc_utils.py]: Same here.

File: src/nupic/math/roc_utils.py (hunk 4 above)
Comment (marionleborgne) [inline src/nupic/math/roc_utils.py]: Thanks! I missed that

File: examples/NuPIC Walkthrough.ipynb (hunk 1 above)
Comment (marionleborgne) [inline examples/NuPIC Walkthrough.ipynb]: I think so (with `BacktrackignTM`). I fixed the notebook, and everything executes properly now.

**Comment by marionleborgne** [comment]
@rhyolight thanks for the feedback. I addressed your comments.

**Comment by rhyolight** [comment]
@marionleborgne Thanks, you'll need to merge.

**Comment by marionleborgne** [comment]
@rhyolight I merged. It's ready for you to review again.

**Pull request merged**